### ISSUE TITLE
Web component: Leaving IBM - There is no label for the modal dialog

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#5060]
## Detailed description

On JAWS FF/Voiceover iPhone
Issue:
There is no label for the modal dialog

Expected:
Provide the heading present in the dialog as the label by associating the same using aria-labelledby

**Comment by andysherman2121** [comment]
@Praveen-sr Can you retest this now that we've applied the solution?

**Comment by Praveen-sr** [comment]
@andysherman2121 Re tested this Web component/React -
Jaws FF
In react wrapper, Screen reader does not identify the dialog label when we open the modal for the first time. But when we refresh the page, the label is identified. 
In web component Canary the label is identified hence working fine.
iPhone VO
Dialog label not identified in both the environments.

**Comment by RobertaJHahn** [comment]
Fixed in Web Components v1.18.0. Needs to be fixed in React.
